aptos-move/framework/aptos-framework/sources/resource_account.move

```diff
@@ -117,21 +122,47 @@ module aptos_framework::resource_account {
     }
 
     /// Creates a new resource account, publishes the package under this account transaction under
-    /// this account and leaves the signer cap readily available for pickup.
+    /// this account, and leaves the signer cap readily available for pickup.
     public entry fun create_resource_account_and_publish_package(
         origin: &signer,
         seed: vector<u8>,
         metadata_serialized: vector<u8>,
         code: vector<vector<u8>>,
-    ) acquires Container {
+    ) acquires BlobStore, Container {
+        create_resource_account_and_publish_package_with_data(origin, seed, metadata_serialized, code, vector::empty());
+    }
+
+    /// Creates a new resource account, publishes the package under this account transaction under
+    /// this account with metadata, and leaves the signer cap readily available for pickup.
+    public entry fun create_resource_account_and_publish_package_with_data(
+        origin: &signer,
+        seed: vector<u8>,
+        metadata_serialized: vector<u8>,
+        code: vector<vector<u8>>,
+        data_blob: vector<u8>,
```

Comment:
What are some examples of data here?

Reply:
@chloeqjz has an "airdrop" NFT contract that would benefit by having the NFT data deployed with the contract without it necessarily being in the contract: https://github.com/aptos-labs/aptos-core/pull/4512

this could also be some sort of initial configuration

the biggest issue is that we don't support deserializing a set of primitive values
just like we can take that tuple in Rust and serialize it
it would be great to support  a deserialization of a tuple in Move
the problem with BCS is that it fails if there are left over bytes
Move provides no means to slice a vector

Reply:
Why not just have the data blob store be a general account feature instead of a resource account specific?

Reply:
Got it now. This data is only meant for one time use, e.g. in init_module